### PR TITLE
fix(hz): base type error and multiple service error for client

### PR DESCRIPTION
**Pull request by FGYFFFF**
#### What type of PR is this?

fix
#### Check the PR title.

- [ ] This PR title match the format: \<type\>(optional scope): \<description\>
- [ ] The description of this PR title is user-oriented and clear enough for others to understand.

#### (Optional) Translate the PR title into Chinese.
修复 hz 生成的 client 代码在使用基本类型作为入参时的错误，以及定义多 service 时出现的错误

#### (Optional) More detail description for this PR(en: English/zh: Chinese).

en: Fix a bug in hz-generated client code when using basic types as input, and when defining multiple services
zh(optional):
修复 hz 生成的 client 代码在使用基本类型作为入参时的错误，以及定义多 service 时出现的错误

### CODE REVIEW

**Review by li-jin-gou — APPROVED** [review]

**Review by Duslia — APPROVED** [review]

File: cmd/hz/generator/package.go [hunk 1]
```diff
@@ -142,7 +142,8 @@ func (pkgGen *HttpPackageGenerator) Generate(pkg *HttpPackage) error {
 
 	if pkgGen.CmdType == meta.CmdClient {
 		clientDir := pkgGen.IdlClientDir
-		clientDir = util.SubDir(clientDir, "hertz")
+		// support multiple service, so disable "clientDir/hertz"
+		// clientDir = util.SubDir(clientDir, "hertz")
```
Comment (welkeyever) [inline cmd/hz/generator/package.go:146]: Why not just delete this line?

File: cmd/hz/generator/client.go [hunk 2]
```diff
@@ -43,7 +43,8 @@ type ClientFile struct {
 
 func (pkgGen *HttpPackageGenerator) genClient(pkg *HttpPackage, clientDir string) error {
 	for _, s := range pkg.Services {
-		hertzClientPath := filepath.Join(clientDir, "hertz_client.go")
+		cliDir := util.SubDir(clientDir, util.ToSnakeCase(s.Name))
+		hertzClientPath := filepath.Join(cliDir, "hertz_client.go")
```
Comment (welkeyever) [inline cmd/hz/generator/client.go:47]: Add a const for `hertz_client.go`?

File: cmd/hz/generator/package.go (hunk 1 above)
Comment (FGYFFFF) [reply cmd/hz/generator/package.go:146]: Do it in another pr, this one has been merged~

File: cmd/hz/generator/client.go (hunk 2 above)
Comment (FGYFFFF) [reply cmd/hz/generator/client.go:47]: done~, in [pr](https://github.com/cloudwego/hertz/pull/511)

File: cmd/hz/generator/package.go (hunk 1 above)
Comment (FGYFFFF) [reply cmd/hz/generator/package.go:146]: done~, in [pr](https://github.com/cloudwego/hertz/pull/511)

**Pull request merged**